packages/effect/src/internal/fiberRuntime.ts

```diff
@@ -1396,6 +1396,23 @@ export const loggerWithConsoleLog = <M, O>(self: Logger<M, O>): Logger<M, void>
     Context.get(services, consoleTag).unsafe.log(self.log(opts))
   })
 
+/** @internal */
+export const loggerWithLeveledLog = <M, O>(self: Logger<M, O>): Logger<M, void> =>
+  internalLogger.makeLogger((opts) => {
+    const services = FiberRefs.getOrDefault(opts.context, defaultServices.currentServices)
+    const unsafeLogger = Context.get(services, consoleTag).unsafe
+    const levels: Partial<Record<LogLevel.LogLevel["_tag"], (v: unknown) => void>> = {
```

Comment:
A `switch` would be better here, instead of creating an intermediary object for every log.